pkg/postage/service.go

```diff
@@ -165,24 +145,36 @@ func (ps *service) IssuerUsable(st *StampIssuer) bool {
 func (ps *service) GetStampIssuer(batchID []byte) (*StampIssuer, func() error, error) {
 	ps.lock.Lock()
 	defer ps.lock.Unlock()
-	for _, st := range ps.issuers {
-		if bytes.Equal(batchID, st.data.BatchID) {
-			if !ps.IssuerUsable(st) {
-				return nil, nil, ErrNotUsable
-			}
-			return st, func() error {
-				return ps.save(st)
-			}, nil
-		}
+
+	if _, ok := ps.issuersInUse[string(batchID)]; ok {
+		return nil, nil, ErrBatchInUse
```

Comment:
We need to also handle this error on the API. In API there are multiple instances of newStamperPutter calls which then handle different types of error. This error needs to be handled separately by sending a detailed error message back to the user that the request failed because another process is using this batch.